Rework trusted proxies

Alkarex opened the pull request:
Fix https://github.com/FreshRSS/FreshRSS/issues/5502
Follow-up of https://github.com/FreshRSS/FreshRSS/pull/3226

New environment variable `TRUSTED_PROXY`: set to 0 to disable, or to a list of trusted IP ranges compatible with https://httpd.apache.org/docs/current/mod/mod_remoteip.html#remoteiptrustedproxy

New internal environment variable `CONN_REMOTE_ADDR` to remember the true IP address of the connection (e.g. last proxy), even when using mod_remoteip.

Current working setups should not observe any significant change.

Alkarex committed:
Rework trusted proxies
Fix https://github.com/FreshRSS/FreshRSS/issues/5502
Follow-up of https://github.com/FreshRSS/FreshRSS/pull/3226

New environment variable `TRUSTED_PROXY`: set to 0 to disable, or to a list of trusted IP ranges compatible with https://httpd.apache.org/docs/current/mod/mod_remoteip.html#remoteiptrustedproxy

New internal environment variable `CONN_REMOTE_ADDR` to remember the true IP address of the connection (e.g. last proxy), even when using mod_remoteip.

Current working setups should not observe any significant change.

Alkarex commented:
To ease testing, I have pushed a Docker image `freshrss/freshrss:pr5549`

Alkarex committed:
Minor whitespace

d-k-bo commented:
From what I can tell, the image works fine on my system, using Caddy's `reverse_proxy` directive without any additional options. I can access the server from an external network and the log still shows my real IP address.

Thank your for working on this!

Alkarex committed:
Safer trusted sources during install
Rework of https://github.com/FreshRSS/FreshRSS/pull/5358
https://github.com/FreshRSS/FreshRSS/issues/5357

Alkarex committed:
Minor readme

joestump commented on lib/lib_rss.php:
```diff
@@ -653,21 +653,43 @@ function checkCIDR(string $ip, string $range): bool {
+	if (empty($trusted)) {
+		$trusted = FreshRSS_Context::$system_conf->trusted_sources;
+	}
+	foreach (FreshRSS_Context::$system_conf->trusted_sources as $cidr) {
```
@Alkarex I've been trying to get this working with Authentik and staring at this logic for a bit. Am I crazy or does this completely override the `$trusted` logic on line 684? 

On line 684 `$trusted` gets set, then if it's `empty()` on line 687 it gets overridden with `FreshRSS_Context::$system_conf->trusted_sources`, but is ignored for the rest of the function as it defers to `FreshRSS_Context::$system_conf->trusted_sources` on this line.

I currently have `TRUSTED_PROXY` set to `192.168.1.34/32`, `$trusted` gets set, and then this falls through to `false` on line 695.

Alkarex replied:
Indeed, this should probably be something like:

```suggestion
	foreach ($trusted as $cidr) {
```

Do you feel like sending a PR?

joestump replied:
@Alkarex that did the trick: https://github.com/FreshRSS/FreshRSS/pull/5853

joestump reviewed (commented):
@Alkarex let me know if you'd like me to file a separate issue on this.
